rfcs/0000-User&HostIntegrationRFC.md

```diff
@@ -0,0 +1,248 @@
+user.profile.department	| keyword |	x256	| Department name associated with the user account.
+user.profile.organization	| keyword |	Elasticsearch Inc.	| Organization name associated with the account.
+user.profile.location	| keyword |	US - Washington - Distributed	| Assigned location for the user account.
+user.profile.mobile_phone	| keyword |	222-222-2222	
+user.profile.primaryPhone	| keyword |	222-222-2222	
+user.profile.secondEmail	| keyword |	first.l@elastic.co	| Additional email addresses associated with the user account.
+user.profile.sup_org_id	| keyword |	SUP-ORG-75	| Primary organization ID for the user account.
+user.profile.supervisory_Org	| keyword |	Field Sales	| Primary organization name for the user account.
+user.profile.assigned_mdm_id	| keyword |	2950	| Primary host owned by the user. This acts as a correlation to identify the host event.			
+user.account.create_date	| date |	June 5, 2023 @ 18:25:57.000	| Date account was created.
+user.account.activated_date	| date |	June 5, 2023 @ 18:25:57.000	| Date account was activated.
+user.account.change_date	| date |	June 5, 2023 @ 18:25:57.000	| Date user account record was last updated at source
+user.account.status.recovery	| boolean |	true/ false	| A flag indicating if account is in recovery
+user.account.status.locked_out	| boolean |	true/ false	| A flag indicating if account is currently locked out
+user.account.status.suspended	| boolean |	true/ false	| A flag indicating if account has been suspended
+user.account.isAdmin	| boolean |	true/ false	| A flag indicating if account is an Admin account
+user.account.isDelegatedAdmin	| boolean |	true/ false	| A flag indicating if account has Delegated Admin rights
+user.account.isPriviledged	| boolean |	true/ false	| A flag indicating if account is a Privileged account
+user.account.status.password_expired	| boolean |	true/ false	| A flag indicating if account password has expired.
+user.account.status.deprovisioned	| boolean |	true/ false	| A flag indicating if account has been deprovisioned
+user.account.password_change_date	| date |	June 5, 2023 @ 18:25:57.000	| Last date/time when account password was updated
+
+### Proposed New Fields for Asset object
+
+Field | Type | Generic Example |	User Entity Example | Host Entity Example | Description
+--- | --- | --- | --- | --- | ---
+asset.category	| keyword |	-	        | Null	                | hardware	                | A further classification of the asset type beyond event.category. For example for host assets {hardware, virtual, container, node}. For user assets {NULL ?}			
+asset.type	    | keyword |	-	        | Null	                | workstation	            | A sub classification of asset. For host assets {workstation, S3, Compute}. For user assets {NULL?}.			
+asset.id	    | keyword |	-	        | 00uhs72c27s6PiK7x1t7	| 2950	                    | A unique ID for the asset. For inventory integrations, it's the id generated from inventory datasource			
```

Comment:
We recommend the `asset ID` be unique within a user's environment. For external directory services, this can map to the directory asset id. 

It will largely depend on the source and management of the data (or how a user configures their external asset management or directory service application).

For the o11y use case you mention above, can this map to the asset.ean field and ensure uniqueness?

Reply:
What we are currently targeting as a model in observability is to use `asset.id` to represent the "internal" ID of that asset, so for example, for a container asset, that container's "Container ID" would be the "asset.id" value. Within all container assets, we expect that to be unique. Pod assets would list their pod ID there, as well. Hosts might use something like an FQDN hostname value there. In some cases, this value may be reasonably unique within the set of assets of that single type, but not unique across assets of other types. For that reason, we have the `asset.ean` value, which at the moment concatenates `asset.kind` + `asset.id` to help namespace the internal asset ID values. The EAN also allows us to expand the convention, URN-style, to include more information in the overall asset's identifier, while still maintaining that asset's own concept of its own ID within its asset space (e.g. container ID).